### DISCUSSION TITLE
How to create record in join table with bulkCreate?

### DISCUSSION THREAD

**Discussion by Siecje** [sequelize/sequelize#16006] [Help & Questions]
I have a join table for students in classes. A student can be in many classes and a class can have many students. A student's name is not unique.

```JavaScript
db['class_student'] = require('./class/class-student')(sequelize, Sequelize);
db.class.belongsToMany(db.student, { through: db.class_student });
db.student.belongsToMany(db.class, { through: db.class_student });
```

I am trying to bulkCreate students and assign classes.

```JavaScript
await db.student.bulkCreate(toCreate, { include: [{ model: db.class_student, as: 'classes' }] });
```

I'm getting an error that
```
name: 'SequelizeEagerLoadingError',
message: 'class_student is not associated to student!'
```

I'm not sure how to create the records in the join table after the `bulkCreate()` since the student names are not unique.

**Comment by ephys** [comment]
You need to define an association between Student & ClassStudent. Sequelize 6 does not do it automatically when you define a BelongsToMany association: https://sequelize.org/docs/v6/advanced-association-concepts/advanced-many-to-many/

Or you can use Sequelize 7, which does it automatically: https://sequelize.org/docs/v7/associations/belongs-to-many/
